File: src/p0013/java/solution.java
class Solution {
    public int romanToInt(String s) {
        int result = 0, prev = 0, curr;
        for (int i = s.length(); i-- > 0; prev = curr) {
            curr = c2n(s.charAt(i));
            if (curr < prev) {
                result -= curr;
            } else {
                result += curr;
            }
        }
        return result;
    }

    private int c2n(char c) {
        switch (c) {
            case 'I':
                return 1;
            case 'V':
                return 5;
            case 'X':
                return 10;
            case 'L':
                return 50;
            case 'C':
                return 100;
            case 'D':
                return 500;
            case 'M':
                return 1000;
            default:
                return 0;
        }
    }
}

File: src/p0013/rust/solution.rs
impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let s = s.as_bytes();
        let mut result = 0;
        let mut prev = 0;
        let mut curr;
        let mut i = s.len();
        while i > 0 {
            i -= 1;
            curr = Self::c2n(s[i]);
            if curr < prev {
                result -= curr;
            } else {
                result += curr;
            }
            prev = curr;
        }
        return result;
    }

    fn c2n(c: u8) -> i32 {
        match c as char {
            'I' => 1,
            'V' => 5,
            'X' => 10,
            'L' => 50,
            'C' => 100,
            'D' => 500,
            'M' => 1000,
            _ => 0
        }
    }
}
